pkgs/modules/haskell/default.nix

```diff
@@ -32,6 +32,6 @@ in
     name = "Haskell Language Server";
     language = "haskell";
 
-    start = "${pkgs.haskell-language-server}/bin/haskell-language-server-wrapper --lsp";
```

Comment:
Have you tested this? It doesn't look like that binary you want to call exists, but maybe it depends on the version?

```
$ ls /nix/store/l24fg4ymhrzwxwlcxh5jmq92sf0g8z2z-haskell-language-server-2.4.0.0/bin/
haskell-language-server-9.4.8  haskell-language-server-wrapper
```